docs/design.md

```diff
@@ -0,0 +1,88 @@
+# Design and architecture
+
+## Design
+The `image-rs` crate is a rustified and tailored version of [containers/image](https://github.com/containers/image),
+to provide a small, simple, secure, lightweight and high performance OCI container image management library for
+[Confidential Containers](https://github.com/confidential-containers).
+
+The `image-rs` crate focuses on usage scenarios to run `Confidential Containers` on Kubernetes clusters with
+[Kata Containers](https://katacontainers.io/). So it implements the
+[CRI Image Service](https://github.com/kubernetes/cri-api/blob/master/pkg/apis/runtime/v1alpha2/api.proto#L119)
+interface to be easily integrated with the K8s ecosystem. Among the container image `Build`, `Ship` and `Run` stages,
+it focuses on the `Run` stage instead of `Build` and `Ship`, that is only to provide interfaces to prepare container
+images for running in a confidential environment. It may also be enhanced to support other confidential container
+runtimes on demand.
+
+The `image-rs` crate supports both encrypted and unencrypted container images since
+[Encrypted Container Image](https://github.com/opencontainers/artifacts/pull/15)
+is mandatory for many confidential container usage scenarios. It also supports image signature verification to ensure
+integrity.
+
+For confidential containers based on enhanced hardware virtualization technologies (TDX/SEV/PEF etc), there's no easy
+way to share container images among pods/containers running on the same physical node. But it's time consuming to
+download, verify, decrypt and extract container images for each pod/container. So the `image-rs` crate will explore
+technologies and innovations to reduce image preparation time. Data decryption and decompression acceleration libraries
+may be used to speed up image decryption and decompression. Image data on demand loading may also be adopted to reduce
+container startup time.
+
+### Goals
+ * Support pulling OCI/docker v2 images from remote registries compatible with the OCI distribution spec.
+ * Support downloading OCI images from local [OCI layout compatiable](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) storage
+ * Support OCI image decryption
+ * Support OCI image signature verification
+ * Support seamlessly integration with Kata Agent and Attestation Agent
+ * Explore image decryption and decompression acceleration technologies
+ * Explore image on demand loading technologies
```

Comment:
I understand that `image-rs` is a crate (library); will it also expose a command-line utility interface?

Reply:
There may be command-line utility for debug, if needed:)

Reply:
Added "Provide a command-line utility for debug"